pkg/agent/inteface.go

```diff
@@ -124,6 +124,14 @@ type InstallStrategy struct {
 
 	// LabelSelector is used to filter clusters based on label. It is only used when strategyType is InstallByLabel
 	LabelSelector *metav1.LabelSelector
+
+	// ManagedClusterFilter will filter the clusters to install the addon to.
+	ManagedClusterFilter func(cluster *clusterv1.ManagedCluster) bool
+}
+
+func (s *InstallStrategy) WithManagedClusterFilter(f func(cluster *clusterv1.ManagedCluster) bool) *InstallStrategy {
```

Comment:
ideally, installAll and labelSelector should also be filter, right?

Reply:
Yes, 
`InstallAll` means filter function always `return true`
`InstallByLabel` means fileter function `return true` when **cluster label matchs**.

Reply:
Not sure whether `LabelSelector ` filed in `InstallStrategy`  is used by other repo.